src/subdomains/staking/domain/entities/staking.entity.ts

```diff
@@ -287,6 +298,69 @@ export class Staking extends IEntity {
     return this.rewardsAmount;
   }
 
+  private validateRewardDistribution(newRewardRoutes: RewardRoute[]): void {
+    const totalDistribution = Util.sumObj<RewardRoute>(newRewardRoutes, 'rewardPercent');
+
+    if (totalDistribution !== 1) {
+      throw new BadRequestException(
+        `Cannot create reward strategy. Total reward distribution must be 100%, instead distributed total of ${Util.round(
+          totalDistribution * 100,
+          2,
+        )}%`,
+      );
+    }
+  }
+
+  private validateDuplicatedRoutes(newRewardRoutes: RewardRoute[]): void {
+    newRewardRoutes.forEach((route) => {
+      const duplicatedRoute = this.findDuplicatedRoute(route, newRewardRoutes);
+
+      if (duplicatedRoute) {
+        throw new BadRequestException(
+          `Cannot create reward strategy. Provided duplicated route for asset ${duplicatedRoute.targetAsset.name} and address ${duplicatedRoute.targetAddress.address}`,
+        );
+      }
+    });
+  }
+
+  private findDuplicatedRoute(currentRoute: RewardRoute, allRewardRoutes: RewardRoute[]): RewardRoute | null {
+    return allRewardRoutes.some(
+      (r, index) =>
+        allRewardRoutes.findIndex(
+          (_r) =>
+            _r.targetAddress.address === r.targetAddress.address &&
+            _r.targetAddress.blockchain === r.targetAddress.blockchain &&
+            _r.targetAsset.id === r.targetAsset.id,
+        ) !== index,
+    )
+      ? currentRoute
+      : null;
+  }
+
+  private updateRewardRoutes(newRewardRoutes: RewardRoute[]): void {
+    this.resetExistingRoutes();
+
+    newRewardRoutes.forEach((newRoute) => {
+      const existingRoute = this.rewardRoutes.find(
+        (_r) =>
+          _r.targetAsset.id === newRoute.targetAsset.id &&
+          _r.targetAddress.address === newRoute.targetAddress.address &&
+          _r.targetAddress.blockchain === newRoute.targetAddress.blockchain,
+      );
+
+      if (existingRoute) {
+        existingRoute.updateRoute(newRoute.label, newRoute.rewardPercent);
+        return;
+      }
+
+      this.rewardRoutes.push(newRoute);
+    });
+  }
+
+  private resetExistingRoutes(): void {
+    this.rewardRoutes.forEach((route) => (route.rewardPercent = 0));
+  }
```

Comment:
yes, lets discuss. I would still return them from API and hide in UI. Cause we might want to allow user seeing previous routes (on separate uncollapse click and greyed out) so that they can reactivate them, instead of creating new one (even from UI point of view)